.github/workflows/build_and_test.yml

```diff
@@ -18,3 +18,57 @@ jobs:
         package-name: rosidl_generator_java rcljava_common rcljava
         source-ros-binary-installation: dashing
         vcs-repo-file-url: https://raw.githubusercontent.com/ros2-java/ros2_java/dashing/ros2_java_desktop.repos
+
+  build_android:
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Install Java
+      run: |
+        sudo apt-get update -qq
+        sudo apt-get install -y default-jdk
+    - name: Setup locale for ROS 2
+      run: |
+        sudo locale-gen en_US en_US.UTF-8
+        sudo update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8
+        export LANG=en_US.UTF-8
+    - name: Setup sources for ROS 2
```

Comment:
I think most (if not all) of the ROS 2 setup can be delegated to the ros-tooling/setup-ros action (similar to the `build_and_test` action above).

Reply:
I started with that, but unfortunately the `ros-tooling` scripts pulled an entire ROS distro, when I only needed `colcon` and just a handful of packages. In any case, I'll investigate more if I can use `ros-tooling` without installing full blown distro, perhaps there's a way, to be honest I didn't look into it enough.

Reply:
ah, yeah. I don't know if there's a way to prevent that action from pulling in a bunch of packages. Never mind then.